lib/rest/routes/pipelines.js

```diff
@@ -0,0 +1,34 @@
+
+const express = require('express');
+
+module.exports = function ({config}) {
+  let router = express.Router();
+  router.put('/:name', function (req, res, next) {
+    config.updateGently(json => {
+      json.pipelines[req.params.name] = req.body;
+      return json;
+    })
+    .then((json) => {
+      res.json({status: 'ok'});
+    })
+    .catch(next);
+  });
+  router.get('/:name', function (req, res, next) {
+    let ep = config.gatewayConfig.pipelines[req.params.name];
+    res.json(ep);
+  });
+  router.get('/', function (req, res, next) {
+    res.json(config.gatewayConfig.pipelines);
+  });
+  router.delete('/:name', function (req, res, next) {
+    config.updateGently(json => {
+      delete json.pipelines[req.params.name];
+      return json;
+    })
+    .then((json) => {
+      res.json({status: 'ok'});
+    })
+    .catch(next);
+  });
+  return router;
```

Comment:
This seems to be almost the same code as `api-endpoints.js` (and other files as well), the only difference is the target json object. It can be probably generalized and reused in all the places you need.

Reply:
no generalization, intentionally copy pasted. 
The reason is further differences: 
validation is one example
additional methods to manipulate policies 

Having some generic monster would greatly affect maintainability